library/alloc/src/boxed.rs

```diff
@@ -1455,9 +1455,35 @@ where
     }
 }
 
+/// Specialization trait used for `From<&[T]>`.
+trait BoxFromSlice<T> {
```

Comment:
This would be dead code in no_global_oom_handling mode.

```suggestion
#[cfg(not(no_global_oom_handling))]
trait BoxFromSlice<T> {
```